src/state/user/hooks.tsx

```diff
@@ -121,15 +124,38 @@ export function useIsAcceptedTerm(): [boolean, (isAcceptedTerm: boolean) => void
 }
 
 export function useDegenModeManager(): [boolean, () => void] {
+  const [swapDegenMode, toggleSwapDegenMode] = useSwapDegenMode()
+  const [poolDegenMode, togglePoolDegenMode] = usePoolDegenMode()
+
+  const { isSwapPage } = usePageLocation()
+  if (isSwapPage) {
```

Comment:
I think no need to check this logic anymore, just use `useSwapDegenMode` in swap pages, and `usePoolDegenMode` in pool pages. You can remove this hook and rename everywhere using it. Don't hesitate to rename it, it's better to be explicit. Or just keep this old hook with the new hook `usePoolDegenMode`, update all pool pages to use new hook.

Reply:
Agree with @XiaoYhun  that explicit is good but this hook is used in too many places. In this case, I prefer to handle selector logic at useSwapDegenMode like what Kien is doing. 
It's the same with what we handled with blackjack.

Reply:
its fine for me then